tenable/ad/profiles/api.py

```diff
@@ -0,0 +1,196 @@
+        Return:
+            list[dict]:
+                The created profile objects
+
+        Example:
+            >>> tad.profiles.create(
+            ...     name='ExampleProfile',
+            ...     directories=[1, 2]
+            ...     )
+        '''
+        payload = [self._schema.dump({
+            'name': name,
+            'directories': directories
+        })]
+
+        return self._schema.load(self._post(json=payload), many=True)
+
+    def details(self,
+                profile_id: str
+                ) -> Dict:
+        '''
+        Retrieves the details for a specific profile
+
+        Args:
+            profile_id (str):
+                The profile instance identifier.
+
+        Returns:
+            dict:
+                The profile object.
+
+        Examples:
+            >>> tad.profiles.details('1')
+        '''
+        return self._schema.load(self._get(f'{profile_id}'))
+
+    def update(self,
+               profile_id: str,
+               **kwargs
+               ) -> Dict:
+        '''
+        Update an existing profile
+
+        Args:
+            profile_id (str):
+                The profile instance identifier.
+            name (optional, str):
+                The name of profile.
+            deleted (optional, bool):
+                is the profile deleted?
+            directories (optional, List[int]):
+                The list of directory ids.
+
+        Returns:
+            dict:
+                The updated profile object.
+
+        Examples:
+            >>> tad.profiles.update(
+            ...     profile_id='1',
+            ...     name='EDITED'
+            ...     )
+        '''
+        payload = self._schema.dump(kwargs)
+        return self._schema.load(self._patch(f"{profile_id}", json=payload))
+
+    def delete(self, profile_id: str) -> None:
+        '''
+        Delete an existing profile
+
+        Args:
+            profile_id (str):
+                The profile instance identifier.
+
+        Returns:
+            None:
+
+        Examples:
+            >>> tad.profiles.delete(profile_id='1')
+        '''
+        self._delete(f"{profile_id}")
+
+    def copy_profile(self,
+                     from_id: str,
+                     name: str,
+                     directories: List[int]
+                     ) -> Dict:
+        '''
+        Creates a new profile from another profile
+
+        Args:
+            from_id (str):
+                The profile instance identifier user wants to copy.
+            name (str):
+                The name of new profile.
+            directories (List[int]):
+                The list of directory ids.
+
+        Returns:
+            dict:
+                The copied role object.
+
+        Examples:
+            >>> tad.profiles.copy_profile(
+            ...     from_id='1',
+            ...     name='Copied name',
+            ...     directories=[1, 2]
+            ...     )
+        '''
+        payload = self._schema.dump({
```

Comment:
should load the schema before dumping to perform validation.

Reply:
update request payload with `schema.load` before dumping.